## Importing Libraries

In [1]:
# importing libraries 
import pandas as pd
pd.options.display.max_columns = None
import numpy as np 
import seaborn as sns 
import matplotlib.pyplot as plt
%matplotlib inline 

In [2]:
# reading dataset 
gamers_df = pd.read_csv('Data/GamingStudy_data.csv', encoding='unicode_escape', index_col=0)

In [3]:
gamers_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13464 entries, 1 to 14250
Data columns (total 54 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Timestamp        13464 non-null  float64
 1   GAD1             13464 non-null  int64  
 2   GAD2             13464 non-null  int64  
 3   GAD3             13464 non-null  int64  
 4   GAD4             13464 non-null  int64  
 5   GAD5             13464 non-null  int64  
 6   GAD6             13464 non-null  int64  
 7   GAD7             13464 non-null  int64  
 8   GADE             12815 non-null  object 
 9   SWL1             13464 non-null  int64  
 10  SWL2             13464 non-null  int64  
 11  SWL3             13464 non-null  int64  
 12  SWL4             13464 non-null  int64  
 13  SWL5             13464 non-null  int64  
 14  Game             13464 non-null  object 
 15  Platform         13464 non-null  object 
 16  Hours            13434 non-null  float64
 17  earnings    

In [4]:
gamers_df.columns

Index(['Timestamp', 'GAD1', 'GAD2', 'GAD3', 'GAD4', 'GAD5', 'GAD6', 'GAD7',
       'GADE', 'SWL1', 'SWL2', 'SWL3', 'SWL4', 'SWL5', 'Game', 'Platform',
       'Hours', 'earnings', 'whyplay', 'League', 'highestleague', 'streams',
       'SPIN1', 'SPIN2', 'SPIN3', 'SPIN4', 'SPIN5', 'SPIN6', 'SPIN7', 'SPIN8',
       'SPIN9', 'SPIN10', 'SPIN11', 'SPIN12', 'SPIN13', 'SPIN14', 'SPIN15',
       'SPIN16', 'SPIN17', 'Narcissism', 'Gender', 'Age', 'Work', 'Degree',
       'Birthplace', 'Residence', 'Reference', 'Playstyle', 'accept', 'GAD_T',
       'SWL_T', 'SPIN_T', 'Residence_ISO3', 'Birthplace_ISO3'],
      dtype='object')

In [5]:
gamers_df.head()

Timestamp  GAD1  GAD2  GAD3  GAD4  GAD5  GAD6  GAD7  \
S. No.                                                          
1       42052.00437     0     0     0     0     1     0     0   
2       42052.00680     1     2     2     2     0     1     0   
3       42052.03860     0     2     2     0     0     3     1   
4       42052.06804     0     0     0     0     0     0     0   
5       42052.08948     2     1     2     2     2     3     2   

                        GADE  SWL1  SWL2  SWL3  SWL4  SWL5    Game  \
S. No.                                                               
1       Not difficult at all     3     5     5     5     5  Skyrim   
2         Somewhat difficult     3     5     2     5     1   Other   
3       Not difficult at all     2     6     5     2     2   Other   
4       Not difficult at all     2     5     5     3     2   Other   
5             Very difficult     2     2     4     5     1   Other   

                       Platform  Hours        earnings     whyplay League  \
S. No.                                                                      
1       Console (PS, Xbox, ...)   15.0  I play for fun  having fun    NaN   
2                            PC    8.0  I play for fun  having fun    NaN   
3                            PC    0.0  I play for fun  having fun    NaN   
4                            PC   20.0  I play for fun   improving    NaN   
5       Console (PS, Xbox, ...)   20.0  I play for fun  having fun    NaN   

        highestleague  streams  SPIN1  SPIN2  SPIN3  SPIN4  SPIN5  SPIN6  \
S. No.                                                                     
1                 NaN      0.0    1.0    0.0    0.0    1.0    0.0    0.0   
2                 NaN      2.0    2.0    1.0    1.0    3.0    2.0    3.0   
3                 NaN      0.0    1.0    0.0    1.0    2.0    3.0    4.0   
4                 NaN      5.0    2.0    0.0    0.0    1.0    3.0    0.0   
5                 NaN      1.0    2.0    0.0    0.0    0.0    1.0    0.0   

        SPIN7  SPIN8  SPIN9  SPIN10  SPIN11  SPIN12  SPIN13  SPIN14  SPIN15  \
S. No.                                                                        
1         1.0    0.0    0.0     0.0     0.0     0.0     0.0     1.0     0.0   
2         1.0    3.0    4.0     0.0     3.0     0.0     1.0     3.0     3.0   
3         2.0    1.0    0.0     1.0     1.0     3.0     0.0     2.0     4.0   
4         0.0    0.0    0.0     0.0     1.0     3.0     0.0     0.0     1.0   
5         0.0    4.0    2.0     0.0     1.0     0.0     0.0     0.0     0.0   

        SPIN16  SPIN17  Narcissism  Gender  Age                       Work  \
S. No.                                                                       
1          1.0     0.0         1.0    Male   25  Unemployed / between jobs   
2          1.0     2.0         1.0    Male   41  Unemployed / between jobs   
3          4.0     2.0         4.0  Female   32                   Employed   
4          0.0     0.0         2.0    Male   28                   Employed   
5          3.0     0.0         1.0    Male   19                   Employed   

                                     Degree Birthplace    Residence Reference  \
S. No.                                                                          
1                  Bachelor (or equivalent)        USA          USA    Reddit   
2                  Bachelor (or equivalent)        USA          USA    Reddit   
3                  Bachelor (or equivalent)    Germany      Germany    Reddit   
4                  Bachelor (or equivalent)        USA          USA    Reddit   
5       High school diploma (or equivalent)        USA  South Korea    Reddit   

                                                Playstyle  accept  GAD_T  \
S. No.                                                                     
1                                            Singleplayer  Accept      1   
2                   Multiplayer - online - with strangers  Accept      8   
3                

The dataset contains information collected on three different psychological scales. For our purposes, we are focusing on using the Generalized Anxiety Disorder (GAD) scale, so we are isolating the GAD scale. To reduce the number of features we have, we will drop `Timestamp`.

In [6]:
# isolating dataset to contain columns of interest 
unnec_cols = ['Timestamp','SWL1', 'SWL2', 'SWL3', 'SWL4', 'SWL5', 'SPIN1', 'SPIN2','SPIN3', 'SPIN4', 'SPIN5','SPIN6','SPIN7', 'SPIN8',
        'SPIN9', 'SPIN10', 'SPIN11','SPIN12', 'SPIN13','SPIN14', 'SPIN15', 'SPIN16', 'SPIN17','Narcissism', 'SWL_T', 
        'SPIN_T','Reference']
gamers_df.drop(columns = unnec_cols, inplace=True)

In [7]:
# viewing the updated dataset
gamers_df.head()

GAD1  GAD2  GAD3  GAD4  GAD5  GAD6  GAD7                  GADE  \
S. No.                                                                   
1          0     0     0     0     1     0     0  Not difficult at all   
2          1     2     2     2     0     1     0    Somewhat difficult   
3          0     2     2     0     0     3     1  Not difficult at all   
4          0     0     0     0     0     0     0  Not difficult at all   
5          2     1     2     2     2     3     2        Very difficult   

          Game                 Platform  Hours        earnings     whyplay  \
S. No.                                                                       
1       Skyrim  Console (PS, Xbox, ...)   15.0  I play for fun  having fun   
2        Other                       PC    8.0  I play for fun  having fun   
3        Other                       PC    0.0  I play for fun  having fun   
4        Other                       PC   20.0  I play for fun   improving   
5        Other  Console (PS, Xbox, ...)   20.0  I play for fun  having fun   

       League  highestleague  streams  Gender  Age                       Work  \
S. No.                                                                          
1         NaN            NaN      0.0    Male   25  Unemployed / between jobs   
2         NaN            NaN      2.0    Male   41  Unemployed / between jobs   
3         NaN            NaN      0.0  Female   32                   Employed   
4         NaN            NaN      5.0    Male   28                   Employed   
5         NaN            NaN      1.0    Male   19                   Employed   

                                     Degree Birthplace    Residence  \
S. No.                                                                
1                  Bachelor (or equivalent)        USA          USA   
2                  Bachelor (or equivalent)        USA          USA   
3                  Bachelor (or equivalent)    Germany      Germany   
4                  Bachelor (or equivalent)        USA          USA   
5       High school diploma (or equivalent)        USA  South Korea   

                                                Playstyle  accept  GAD_T  \
S. No.                                                                     
1                                            Singleplayer  Accept      1   
2                   Multiplayer - online - with strangers  Accept      8   
3                                            Singleplayer  Accept      8   
4       Multiplayer - online - with online acquaintanc...  Accept      0   
5                   Multiplayer - online - with strangers  Accept     14   

       Residence_ISO3 Birthplace_ISO3  
S. No.                                 
1                 USA             USA  
2                 USA             USA  
3                 DEU             DEU  
4                 USA             USA  
5                 KOR             USA

Now that we've isolated our columns of interest, let's break down the structure and see what we're working with. 

In [8]:
# viewing nans in our dataset 
gamers_df.isna().sum()

GAD1                   0
GAD2                   0
GAD3                   0
GAD4                   0
GAD5                   0
GAD6                   0
GAD7                   0
GADE                 649
Game                   0
Platform               0
Hours                 30
earnings               0
whyplay                0
League              1838
highestleague      13464
streams              100
Gender                 0
Age                    0
Work                  38
Degree                 0
Birthplace             0
Residence              0
Playstyle              0
accept               414
GAD_T                  0
Residence_ISO3       110
Birthplace_ISO3      121
dtype: int64

It looks like `GADE`, `Hours`, `League`, `highestleague`, `streams`, `accept`, `Residence_ISO3`, and `Birthplace_ISO3` contain missing or NaN values, so we will need to look at them more closely. 

### Survey Participant Consent 
To ensure participant privacy, we need to evaluate whether there are any records/participants that did not complete the consent form. 

In [9]:
# viewing if any missing data regarding participant consent 
gamers_df['accept'].isna().sum()

414

There are 414 missing records where participant consent is unknown. For personal privacy reasons, we are dropping them from the dataset. 

In [10]:
# dropping missing records where participant consent is unknown 
gamers_df.dropna(subset = 'accept', inplace=True)

In [11]:
# verifying records were dropped 
gamers_df['accept'].isna().sum()

0

### Generalized Anxiety Disorder (GAD-7) Items 

In [12]:
# viewing GAD scale items 
gad_cols = ['GAD1', 'GAD2', 'GAD3', 'GAD4', 'GAD5', 'GAD6', 'GAD7', 'GADE', 'GAD_T']
gamers_df.loc[:, gad_cols].isna().sum()

GAD1       0
GAD2       0
GAD3       0
GAD4       0
GAD5       0
GAD6       0
GAD7       0
GADE     631
GAD_T      0
dtype: int64

In [13]:
gamers_df['GADE'].value_counts()

Not difficult at all    6040
Somewhat difficult      4982
Very difficult           994
Extremely difficult      403
Name: GADE, dtype: int64

Based on the survey used in the dataset, `GADE` refers to responses to an optional item: **"If you checked off any problems, how difficult have they made it for you to do your work, take care of things at home, or get along with other people?"** In later model preprocessing steps, we shall impute Nans after train-test splitting with 'Did Not Answer'.

### League

Both `League` and `highestleague` are game-specific rankings where participants filled in responses based on the type of game they most frequently played. 

In [14]:
league_cols = ['League','highestleague']
gamers_df.loc[:,league_cols].isna().sum()

League            1786
highestleague    13050
dtype: int64

In [15]:
gamers_df['League'].value_counts()

Gold                                                 939
Silver                                               637
Platinum                                             621
Diamond                                              525
gold                                                 305
                                                    ... 
in cs:go double ak, in league gold                     1
Silver 1 (CS) Silver 5 (lol)                           1
Not played ranked yet, was Platinum 2 last season      1
Top team(DP2), Plat 3(LoL)                             1
AHGL                                                   1
Name: League, Length: 1425, dtype: int64

In [16]:
gamers_df['highestleague'].unique()

array([nan])

`highestleague` only contains NaN values. 

In [17]:
gamers_df['League'].isna().sum()

1786

`League` and `highestleague` are game-specific features and it is difficult to conclude whether game-specific rankings are generalizable across different games, so we are choosing to drop both features. Our goal is to limit biases towards any specific game based on the competitive ranking system they may use. Future work could look to ascertaining whether gaming habits in high-stakes competitive environments have a relationship with different psychometric scales.   

In [18]:
# dropping League and highestleague from the dataset 
gamers_df.drop(columns = ['League','highestleague'], inplace=True)

In [19]:
# verifying whether columns were dropped
gamers_df

GAD1  GAD2  GAD3  GAD4  GAD5  GAD6  GAD7                  GADE  \
S. No.                                                                   
1          0     0     0     0     1     0     0  Not difficult at all   
2          1     2     2     2     0     1     0    Somewhat difficult   
3          0     2     2     0     0     3     1  Not difficult at all   
4          0     0     0     0     0     0     0  Not difficult at all   
5          2     1     2     2     2     3     2        Very difficult   
...      ...   ...   ...   ...   ...   ...   ...                   ...   
14246      1     0     0     1     0     1     1  Not difficult at all   
14247      3     3     3     3     2     3     3   Extremely difficult   
14248      0     0     0     0     0     0     0  Not difficult at all   
14249      3     2     1     3     0     1     3    Somewhat difficult   
14250      1     1     0     0     0     0     0    Somewhat difficult   

                     Game                 Platform  Hours        earnings  \
S. No.                                                                      
1                  Skyrim  Console (PS, Xbox, ...)   15.0  I play for fun   
2                   Other                       PC    8.0  I play for fun   
3                   Other                       PC    0.0  I play for fun   
4                   Other                       PC   20.0  I play for fun   
5                   Other  Console (PS, Xbox, ...)   20.0  I play for fun   
...                   ...                      ...    ...             ...   
14246               Other  Console (PS, Xbox, ...)   40.0  I play for fun   
14247   League of Legends                       PC   20.0  I play for fun   
14248         Starcraft 2                       PC   20.0  I play for fun   
14249               Other  Console (PS, Xbox, ...)    5.0  I play for fun   
14250         Starcraft 2                       PC   15.0  I play for fun   

           whyplay  streams  Gender  Age                             Work  \
S. No.                                                                      
1       having fun      0.0    Male   25        Unemployed / between jobs   
2       having fun      2.0    Male   41        Unemployed / between jobs   
3       having fun      0.0  Female   32                         Employed   
4        improving      5.0    Male   28                         Employed   
5       having fun      1.0    Male   19                         Employed   
...            ...      ...     ...  ...                              ...   
14246    improving     10.0    Male   22  Student at college / university   
14247      winning     20.0  Female   20  Student at college / university   
14248   having fun     14.0    Male   19  Student at college / university   
14249   having fun      0.0  Female   18                Student at school   
14250    improving      8.0    Male   19  Student at college / university   

                                     Degree Birthplace    Residence  \
S. No.                                                                
1                  Bachelor (or equivalent)        USA          USA   
2                  Bachelor (or equivalent)        USA          USA   
3                  Bachelor (or equivalent)    Germany      Germany   
4                  Bachelor (or equivalent)        USA          USA   
5       High school diploma (or equivalent)        USA  South Korea   
...                                     ...        ...          ...   
14246                Master (or equivalent)     France       France   
14247   High school diploma (or equivalent)        USA          USA   
14248   High school diploma (or equivalent)     Norway       Norway   
14249   High school diploma (or equivalent)     Canada       Canada   
14250   High school diploma (or equivalent)     Canada       Canada   

                                                Playstyle  accept  GAD_T  \
S. No.                                   

### Streams

`streams` refers to responses to the following question: "In addition to playing, how many hours a week do you spend on watching online player/tournament streams/replays/demos, giving or receiving coaching?"

In [20]:
gamers_df['streams'].isna().sum()

95

In [21]:
gamers_df['streams'].describe()

count    12955.000000
mean        11.269857
std         79.763804
min          0.000000
25%          4.000000
50%          8.000000
75%         15.000000
max       9001.000000
Name: streams, dtype: float64

NaN values in `streams` may indicate that no additional hours were spent aside from gaming to the above exemplars. However, based on preliminary EDA, values are heavily imbalanced, something to consider for future preprocessing steps. 

### Participant Residence & Birthplace

In [22]:
geo_loc = ['Birthplace_ISO3','Residence_ISO3']
gamers_df.loc[:,geo_loc]

Birthplace_ISO3 Residence_ISO3
S. No.                               
1                  USA            USA
2                  USA            USA
3                  DEU            DEU
4                  USA            USA
5                  USA            KOR
...                ...            ...
14246              FRA            FRA
14247              USA            USA
14248              NOR            NOR
14249              CAN            CAN
14250              CAN            CAN

[13050 rows x 2 columns]

In [23]:
gamers_df[gamers_df['Residence_ISO3'].isna()].head()

GAD1  GAD2  GAD3  GAD4  GAD5  GAD6  GAD7                  GADE  \
S. No.                                                                   
180        0     2     0     0     0     3     0  Not difficult at all   
376        0     1     1     0     0     0     0  Not difficult at all   
462        0     0     0     0     0     0     0  Not difficult at all   
725        3     3     2     2     3     3     0  Not difficult at all   
797        1     0     0     0     1     2     1  Not difficult at all   

                     Game Platform  Hours        earnings     whyplay  \
S. No.                                                                  
180     League of Legends       PC    7.0  I play for fun   improving   
376           Starcraft 2       PC   14.0  I play for fun  having fun   
462     League of Legends       PC   40.0  I play for fun   improving   
725     League of Legends       PC   17.0  I play for fun  having fun   
797     League of Legends       PC   35.0  I play for fun     winning   

        streams Gender  Age                             Work  \
S. No.                                                         
180         NaN   Male   19  Student at college / university   
376         0.0   Male   35                         Employed   
462        10.0   Male   24                         Employed   
725         8.0   Male   20        Unemployed / between jobs   
797        10.0   Male   18                Student at school   

                                     Degree Birthplace Residence  \
S. No.                                                             
180                                    None    Unknown   Unknown   
376                  Master (or equivalent)    Germany   Unknown   
462     High school diploma (or equivalent)    Unknown   Unknown   
725                                    None   Portugal   Unknown   
797     High school diploma (or equivalent)  Lithuania   Unknown   

                                            Playstyle  accept  GAD_T  \
S. No.                                                                 
180     Multiplayer - online - with real life friends  Accept      5   
376             Multiplayer - online - with strangers  Accept      2   
462             Multiplayer - online - with strangers  Accept      0   
725     Multiplayer - online - with real life friends  Accept     16   
797             Multiplayer - online - with strangers  Accept      5   

       Residence_ISO3 Birthplace_ISO3  
S. No.                                 
180               NaN             NaN  
376               NaN             DEU  
462               NaN             NaN  
725               NaN             PRT  
797               NaN             LTU

In [24]:
gamers_df[gamers_df['Residence'] == 'Unknown']

GAD1  GAD2  GAD3  GAD4  GAD5  GAD6  GAD7                  GADE  \
S. No.                                                                   
180        0     2     0     0     0     3     0  Not difficult at all   
376        0     1     1     0     0     0     0  Not difficult at all   
462        0     0     0     0     0     0     0  Not difficult at all   
725        3     3     2     2     3     3     0  Not difficult at all   
797        1     0     0     0     1     2     1  Not difficult at all   
...      ...   ...   ...   ...   ...   ...   ...                   ...   
13723      1     1     0     1     0     2     2  Not difficult at all   
13879      1     0     0     0     0     3     0    Somewhat difficult   
13950      2     3     2     2     3     2     2  Not difficult at all   
14056      0     0     0     0     0     3     0   Extremely difficult   
14105      0     0     0     1     1     1     1    Somewhat difficult   

                     Game Platform  Hours        earnings     whyplay  \
S. No.                                                                  
180     League of Legends       PC    7.0  I play for fun   improving   
376           Starcraft 2       PC   14.0  I play for fun  having fun   
462     League of Legends       PC   40.0  I play for fun   improving   
725     League of Legends       PC   17.0  I play for fun  having fun   
797     League of Legends       PC   35.0  I play for fun     winning   
...                   ...      ...    ...             ...         ...   
13723   League of Legends       PC   11.0  I play for fun   improving   
13879   League of Legends       PC   30.0  I play for fun   improving   
13950   League of Legends       PC   15.0  I play for fun   improving   
14056   League of Legends       PC  150.0  I play for fun   improving   
14105   League of Legends       PC   70.0  I play for fun   improving   

        streams Gender  Age                             Work  \
S. No.                                                         
180         NaN   Male   19  Student at college / university   
376         0.0   Male   35                         Employed   
462        10.0   Male   24                         Employed   
725         8.0   Male   20        Unemployed / between jobs   
797        10.0   Male   18                Student at school   
...         ...    ...  ...                              ...   
13723       2.0   Male   22  Student at college / university   
13879       6.0   Male   19  Student at college / university   
13950       0.0   Male   18  Student at college / university   
14056       8.0   Male   18                Student at school   
14105       5.0   Male   24  Student at college / university   

                                     Degree Birthplace Residence  \
S. No.                                                             
180                                    None    Unknown   Unknown   
376                  Master (or equivalent)    Germany   Unknown   
462     High school diploma (or equivalent)    Unknown   Unknown   
725                                    None   Portugal   Unknown   
797     High school diploma (or equivalent)  Lithuania   Unknown   
...                                     ...        ...       ...   
13723   High school diploma (or equivalent)    Unknown   Unknown   
13879                                  None    Germany   Unknown   
13950   High school diploma (or equivalent)    Unknown   Unknown   
14056   High school diploma (or equivalent)  Australia   Unknown   
14105   High school diploma (or equivalent)     Cyprus   Unknown   

                                                Playstyle  accept  GAD_T  \
S. No.                                                                     
180         Multiplayer - online - with real life friends  Accept      5   
376                 Multiplayer - online - with strangers  Accept      2   
462                 Multiplayer - online - with strangers  Accept      0 

In [25]:
# viewing NaN values in the Residence column
gamers_df['Residence_ISO3'].isna().sum()

104

In [26]:
gamers_df[gamers_df['Residence_ISO3'].isna()].head()

GAD1  GAD2  GAD3  GAD4  GAD5  GAD6  GAD7                  GADE  \
S. No.                                                                   
180        0     2     0     0     0     3     0  Not difficult at all   
376        0     1     1     0     0     0     0  Not difficult at all   
462        0     0     0     0     0     0     0  Not difficult at all   
725        3     3     2     2     3     3     0  Not difficult at all   
797        1     0     0     0     1     2     1  Not difficult at all   

                     Game Platform  Hours        earnings     whyplay  \
S. No.                                                                  
180     League of Legends       PC    7.0  I play for fun   improving   
376           Starcraft 2       PC   14.0  I play for fun  having fun   
462     League of Legends       PC   40.0  I play for fun   improving   
725     League of Legends       PC   17.0  I play for fun  having fun   
797     League of Legends       PC   35.0  I play for fun     winning   

        streams Gender  Age                             Work  \
S. No.                                                         
180         NaN   Male   19  Student at college / university   
376         0.0   Male   35                         Employed   
462        10.0   Male   24                         Employed   
725         8.0   Male   20        Unemployed / between jobs   
797        10.0   Male   18                Student at school   

                                     Degree Birthplace Residence  \
S. No.                                                             
180                                    None    Unknown   Unknown   
376                  Master (or equivalent)    Germany   Unknown   
462     High school diploma (or equivalent)    Unknown   Unknown   
725                                    None   Portugal   Unknown   
797     High school diploma (or equivalent)  Lithuania   Unknown   

                                            Playstyle  accept  GAD_T  \
S. No.                                                                 
180     Multiplayer - online - with real life friends  Accept      5   
376             Multiplayer - online - with strangers  Accept      2   
462             Multiplayer - online - with strangers  Accept      0   
725     Multiplayer - online - with real life friends  Accept     16   
797             Multiplayer - online - with strangers  Accept      5   

       Residence_ISO3 Birthplace_ISO3  
S. No.                                 
180               NaN             NaN  
376               NaN             DEU  
462               NaN             NaN  
725               NaN             PRT  
797               NaN             LTU

In [27]:
gamers_df[gamers_df['Birthplace_ISO3'].isna()].head()

GAD1  GAD2  GAD3  GAD4  GAD5  GAD6  GAD7                  GADE  \
S. No.                                                                   
180        0     2     0     0     0     3     0  Not difficult at all   
340        0     0     1     0     0     1     1  Not difficult at all   
370        0     0     1     1     0     0     0    Somewhat difficult   
426        1     0     1     2     1     1     0    Somewhat difficult   
462        0     0     0     0     0     0     0  Not difficult at all   

                     Game Platform  Hours  \
S. No.                                      
180     League of Legends       PC    7.0   
340     League of Legends       PC   25.0   
370           Starcraft 2       PC   20.0   
426     League of Legends       PC   20.0   
462     League of Legends       PC   40.0   

                                                 earnings  \
S. No.                                                      
180                                        I play for fun   
340                                        I play for fun   
370                                        I play for fun   
426     I play mostly for fun but earn a little on the...   
462                                        I play for fun   

                         whyplay  streams Gender  Age  \
S. No.                                                  
180                    improving      NaN   Male   19   
340                      winning     10.0   Male   24   
370                    improving     20.0   Male   21   
426     having fun and improving      5.0   Male   19   
462                    improving     10.0   Male   24   

                                   Work                               Degree  \
S. No.                                                                         
180     Student at college / university                                 None   
340                            Employed               Master (or equivalent)   
370     Student at college / university             Bachelor (or equivalent)   
426     Student at college / university  High school diploma (or equivalent)   
462                            Employed  High school diploma (or equivalent)   

       Birthplace    Residence  \
S. No.                           
180       Unknown      Unknown   
340       Unknown      Ukraine   
370       Unknown           UK   
426       Unknown  Netherlands   
462       Unknown      Unknown   

                                                Playstyle  accept  GAD_T  \
S. No.                                                                     
180         Multiplayer - online - with real life friends  Accept      5   
340         Multiplayer - online - with real life friends  Accept      3   
370                 Multiplayer - online - with strangers  Accept      2   
426     Multiplayer - online - with online acquaintanc...  Accept      6   
462                 Multiplayer - online - with strangers  Accept      0   

       Residence_ISO3 Birthplace_ISO3  
S. No.                                 
180               NaN             NaN  
340               UKR             NaN  
370               GBR             NaN  
426               NLD             NaN  
462               NaN             NaN

It seems that where `Residence_ISO3` and `Birthplace_ISO3` are NaN, the related columns of `Birthplace` and `Residence` are both 'Unknown'. This could be due to participant personal privacy reasons or dataset errors. For future preprocessing steps, we will use the standardized ISO 3166-1 alpha-3 format and use the value 'AAA' to impute any NaN values.

### Hours  

`Hours` refers to the number of hours played per week. Survey responses are game-specific and relate to hours played of the value in `Game`.

In [28]:
gamers_df['Hours'].isna().sum()

27

In [29]:
gamers_df['Hours'].value_counts().sort_index(ascending=True)

0.0        11
1.0        22
2.0        70
3.0       101
4.0       140
         ... 
140.0       2
150.0       1
160.0       1
420.0       1
8000.0      1
Name: Hours, Length: 84, dtype: int64

In [30]:
# viewing NaN values in Hours column
gamers_df[gamers_df['Hours'].isna()]

GAD1  GAD2  GAD3  GAD4  GAD5  GAD6  GAD7                  GADE  \
S. No.                                                                   
46         2     1     3     1     0     1     0    Somewhat difficult   
1316       1     2     2     2     1     1     3   Extremely difficult   
1488       0     1     1     0     0     0     1    Somewhat difficult   
1972       3     3     3     3     0     2     3    Somewhat difficult   
2350       0     0     0     0     0     0     0                   NaN   
2539       1     2     1     2     1     0     1    Somewhat difficult   
3259       1     1     1     1     1     1     1  Not difficult at all   
4336       1     1     2     0     1     0     0                   NaN   
5232       3     3     3     3     3     3     3   Extremely difficult   
5518       3     1     1     1     0     2     2    Somewhat difficult   
6778       0     0     0     0     0     2     0  Not difficult at all   
6787       0     0     0     0     0     1     0  Not difficult at all   
6906       3     1     2     3     0     3     0    Somewhat difficult   
7873       0     0     0     1     0     0     1  Not difficult at all   
8271       0     0     0     0     1     1     0  Not difficult at all   
8756       3     3     3     3     0     1     2    Somewhat difficult   
8813       3     1     1     1     0     2     2        Very difficult   
8900       1     1     1     1     0     1     0    Somewhat difficult   
8974       3     2     1     1     1     3     2        Very difficult   
9259       3     3     3     3     3     3     3    Somewhat difficult   
10339      2     2     2     0     3     2     0   Extremely difficult   
10398      3     2     2     2     3     0     3   Extremely difficult   
10569      1     0     2     0     1     0     0  Not difficult at all   
11414      1     0     1     0     1     1     0  Not difficult at all   
13335      3     1     3     3     3     3     3   Extremely difficult   
13451      0     0     0     0     0     1     0  Not difficult at all   
13484      2     2     2     1     2     0     1        Very difficult   

                     Game                 Platform  Hours  \
S. No.                                                      
46         Counter Strike                       PC    NaN   
1316    League of Legends                       PC    NaN   
1488    League of Legends                       PC    NaN   
1972    League of Legends                       PC    NaN   
2350       Counter Strike                       PC    NaN   
2539    League of Legends                       PC    NaN   
3259    League of Legends                       PC    NaN   
4336    League of Legends                       PC    NaN   
5232                Other                       PC    NaN   
5518    League of Legends                       PC    NaN   
6778                Other                       PC    NaN   
6787    League of Legends                       PC    NaN   
6906    League of Legends                       PC    NaN   
7873    League of Legends                       PC    NaN   
8271    League of Legends                       PC    NaN   
8756                Other                       PC    NaN   
8813    League of Legends                       PC    NaN   
8900    League of Legends                       PC    NaN   
8974                Other                       PC    NaN   
9259    League of Legends                       PC    NaN   
10339   League of Legends                       PC    NaN   
10398   League of Legends                       PC    NaN   
10569   League of Legends                       PC    NaN   
11414   League of Legends                       PC    NaN   
13335               Other  Console (PS, Xbox, ...)    NaN   
13451   League of Legends                       PC    NaN   
13484   League of Legends                       PC    NaN   

                                                 earnings  \
S. No.                        

Where `Hours` contains NaN values, the common `Game` played are "Counter Strike, League of Legends, and Other". For future preprocessing steps, we will consider multivariate feature imputation based on the game played. 

## Non-NaN Column Data Wrangling

Now that we have a rough idea of how NaN values are distributed amongst the columns containing them, let's move on and see if any other columns need cleaning. As a reference, we will use the survey design to see where cleaning may be needed. 

In [31]:
gamers_df.tail()

GAD1  GAD2  GAD3  GAD4  GAD5  GAD6  GAD7                  GADE  \
S. No.                                                                   
14246      1     0     0     1     0     1     1  Not difficult at all   
14247      3     3     3     3     2     3     3   Extremely difficult   
14248      0     0     0     0     0     0     0  Not difficult at all   
14249      3     2     1     3     0     1     3    Somewhat difficult   
14250      1     1     0     0     0     0     0    Somewhat difficult   

                     Game                 Platform  Hours        earnings  \
S. No.                                                                      
14246               Other  Console (PS, Xbox, ...)   40.0  I play for fun   
14247   League of Legends                       PC   20.0  I play for fun   
14248         Starcraft 2                       PC   20.0  I play for fun   
14249               Other  Console (PS, Xbox, ...)    5.0  I play for fun   
14250         Starcraft 2                       PC   15.0  I play for fun   

           whyplay  streams  Gender  Age                             Work  \
S. No.                                                                      
14246    improving     10.0    Male   22  Student at college / university   
14247      winning     20.0  Female   20  Student at college / university   
14248   having fun     14.0    Male   19  Student at college / university   
14249   having fun      0.0  Female   18                Student at school   
14250    improving      8.0    Male   19  Student at college / university   

                                     Degree Birthplace Residence  \
S. No.                                                             
14246                Master (or equivalent)     France    France   
14247   High school diploma (or equivalent)        USA       USA   
14248   High school diploma (or equivalent)     Norway    Norway   
14249   High school diploma (or equivalent)     Canada    Canada   
14250   High school diploma (or equivalent)     Canada    Canada   

                                            Playstyle  accept  GAD_T  \
S. No.                                                                 
14246           Multiplayer - online - with strangers  Accept      4   
14247           Multiplayer - online - with strangers  Accept     20   
14248   Multiplayer - online - with real life friends  Accept      0   
14249                                    Singleplayer  Accept     13   
14250           Multiplayer - online - with strangers  Accept      2   

       Residence_ISO3 Birthplace_ISO3  
S. No.                                 
14246             FRA             FRA  
14247             USA             USA  
14248             NOR             NOR  
14249             CAN             CAN  
14250             CAN             CAN

### Game Most Frequently Played

In [32]:
# seeing if any NaN values in Game column 
gamers_df['Game'].isna().sum()

0

In [33]:
# distribution of games most frequently played 
gamers_df['Game'].value_counts()

League of Legends      10961
Other                    992
Starcraft 2              334
Counter Strike           305
World of Warcraft        147
Hearthstone               99
Diablo 3                  88
Heroes of the Storm       41
Guild Wars 2              37
Skyrim                    28
Destiny                   18
Name: Game, dtype: int64

There are no NaN values in `Game` column, so no data cleaning steps are needed.

### Gaming Platform Used   

`Platform` relates to what gaming console or platform the participants reported using to play the game they play themost currently. 

In [34]:
gamers_df['Platform'].isna().sum()

0

In [35]:
gamers_df['Platform'].value_counts()

PC                         12813
Console (PS, Xbox, ...)      214
Smartphone / Tablet           23
Name: Platform, dtype: int64

### Earnings  

In [36]:
gamers_df['earnings'].isna().sum()

0

In [37]:
gamers_df['earnings'].value_counts()

I play for fun                                                                                           11742
I play mostly for fun but earn a little on the side (tournament winnings, streaming, etc)                  966
I earn a living by playing this game                                                                        45
I play to win                                                                                                3
i try                                                                                                        2
                                                                                                         ...  
Keep in touch w/ friends                                                                                     1
I Play for fun and working on making earn out of it.                                                         1
I play ranked mostly                                                                                         1
I

In [38]:
earnings_answers = ['I earn a living by playing this game', 'I play for fun', 
                    'I play mostly for fun but earn a little on the side (tournament winnings, streaming, etc)',
                   'I play to win']
gamers_df = gamers_df[gamers_df['earnings'].isin(earnings_answers)]

In [39]:
gamers_df['earnings'].value_counts()

I play for fun                                                                               11742
I play mostly for fun but earn a little on the side (tournament winnings, streaming, etc)      966
I earn a living by playing this game                                                            45
I play to win                                                                                    3
Name: earnings, dtype: int64

`earnings` contains answers to the following question: "Is playing this game your hobby or do you make money from it?". Answers can be grouped into three multiple choice answers (a) I earn a living by playing this game, (b) I play mostly for fun but earn a little on the side (tournament winnings, streaming,etc), (c) I play for fun and an optional fill-in response. There are 311 fill-ins, so we will drop them in future data preprocessing steps. 

### Reason for Playing  

In [40]:
gamers_df['whyplay'].value_counts()

having fun                                                                          5115
improving                                                                           4612
winning                                                                             1978
relaxing                                                                             617
All of the above                                                                      23
                                                                                    ... 
Improving + Fun                                                                        1
to be occupied                                                                         1
Improving, having fun & winning                                                        1
it just works as a distraction. other then that winning and climbing that ladder       1
Mental priority is Fun>Improve>Win, actual priority once in game Win>Improve>Fun       1
Name: whyplay, Length

In [41]:
gamers_df['whyplay'].replace('all of the above'.lower(), 'All of the above'.lower(), inplace=True)

C:\Users\Kevin\AppData\Local\Temp\ipykernel_11160\75035263.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  gamers_df['whyplay'].replace('all of the above'.lower(), 'All of the above'.lower(), inplace=True)


In [42]:
gamers_df['whyplay'].value_counts()

having fun                                                                          5115
improving                                                                           4612
winning                                                                             1978
relaxing                                                                             617
All of the above                                                                      23
                                                                                    ... 
Improving + Fun                                                                        1
to be occupied                                                                         1
Improving, having fun & winning                                                        1
it just works as a distraction. other then that winning and climbing that ladder       1
Mental priority is Fun>Improve>Win, actual priority once in game Win>Improve>Fun       1
Name: whyplay, Length

In [44]:
reason_play = ['winning', 'improving', 'relaxing', 'having fun']
gamers_df = gamers_df[gamers_df['whyplay'].isin(reason_play)]

### Gender

In [45]:
gamers_df['Gender'].isna().sum()

0

In [46]:
gamers_df['Gender'].value_counts()

Male      11608
Female      669
Other        45
Name: Gender, dtype: int64

### Age  

In [47]:
gamers_df['Age'].isna().sum()

0

In [48]:
gamers_df['Age'].value_counts()

18    3272
19    1971
20    1691
21    1379
22    1045
23     796
24     602
25     479
26     314
27     214
28     146
29     111
30      82
31      52
32      47
33      34
34      18
35      17
36      16
38       8
39       5
37       5
42       5
40       3
41       3
50       2
44       2
63       1
49       1
56       1
Name: Age, dtype: int64

### Playstyle 

In [49]:
gamers_df['Playstyle'].isna().sum()

0

In [50]:
gamers_df['Playstyle'].value_counts()

Multiplayer - online - with real life friends                                   5201
Multiplayer - online - with strangers                                           3771
Multiplayer - online - with online acquaintances or teammates                   2407
Singleplayer                                                                     689
Multiplayer - offline (people in the same room)                                   43
                                                                                ... 
Multiplayer equally with strangers or real life friends                            1
Multiplayer - online - with strangers against A.I.                                 1
Online with friends, both real life friends and friends I've made online.          1
All multiplayer options listed above.                                              1
I usually play online with my real life friends and the internet ones alike.       1
Name: Playstyle, Length: 210, dtype: int64

In [51]:
playstyle_resp = ['Singleplayer', 'Multiplayer - offline (people in the same room)',
                 'Multiplayer - online - with strangers', 'Multiplayer - online - with online acquaintances or teammates', 
                 'Multiplayer - online - with real life friends']
gamers_df['Playstyle'].isin(playstyle_resp).value_counts()

True     12111
False      211
Name: Playstyle, dtype: int64

In [53]:
gamers_df = gamers_df[gamers_df['Playstyle'].isin(playstyle_resp)]

### Duplicates 

In [54]:
# dropping duplicate records
gamers_df.drop_duplicates(inplace=True)

## Exporting Cleaned Data

In [74]:
# dropping 'accept' column now that we've dropped records with missing participant information 
gamers_df.drop(columns = ['accept'], inplace=True)

In [80]:
# exporting data to new csv file 
gamers_df.to_csv('Data/cleaned_data')

## Conclusion 

In this section, we have started to develop an understanding of the data and cleaned up the values in our dataset. We've reduced the dataset to contain information on participants' demographics, gaming habits, and narrowed our focus to identify items related to the Generalized Anxiety Disorder Screener (GAD-7). We've removed `Timestamp` as well as missing values from the `accept` column to maintain participant privacy. We've also removed `Leage` and `highestleague` from the dataset to maintain data coherency for our future model. 

We've also outlined some steps for future data preprocessing steps (i.e., multivariate imputation strategies and filling in NaN values). In EDA, we will continue to explore the data through visualizations and prepare data for model preprocessing. 